### PR TITLE
Added Lock Options

### PR DESCRIPTION
**Pull request by bbernaz**

### CODE REVIEW

File: src/views/Swap/SwapGuru/index.tsx [hunk 1]
```diff
@@ -354,42 +356,22 @@ function SwapGuru() {
...
-                          style: {
-                            fontSize: "20px",
-                            lineHeight: "25px",
-                            fontWeight: 400,
-                          },
-                        }}
-                      />
-                    </MenuItem>
+                    {lockOptions.map((option: any) => {
+                      return (
```
Comment (jiggyBu) [inline src/views/Swap/SwapGuru/index.tsx]: nit: no need for return statement

File: src/views/Swap/SwapGuru/index.tsx [hunk 2]
```diff
@@ -354,42 +356,22 @@ function SwapGuru() {
...
+                        <MenuItem value={option.value} className={classes.menuItem}>
+                          <ListItemText
+                            primary={option.name}
+                            primaryTypographyProps={{
+                              style: {
```
Comment (jiggyBu) [inline src/views/Swap/SwapGuru/index.tsx]: Hmm, could we define with `createStyles` as https://github.com/NidhiDAO/NidhiDAO-frontend/pull/24/files#diff-ecc47e96602cfa104386c81c4b329be4697cf1dfd071fba3bab681a0e69c2103R361 and use it in here

File: src/views/Swap/useLockOptions.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  React.useEffect(() => {
+    if (chainID && provider) {
+      (async function () {
+        const options = [];
+        const contract = new ethers.Contract(addresses[chainID].PASSIVE_INCOME_NFT, PassiveIncomeNFT, provider);
+        const maxLockDuration = await contract.maxLockDuration();
+
+        for (let index = Number(MIN_LOCK_DURATION); index <= maxLockDuration; index++) {
+          options.push({ name: `${index} Months`, value: index });
```
Comment (jiggyBu) [inline src/views/Swap/useLockOptions.ts]:
```suggestion
          options.push({ name: `${index} months`, value: index });
```

File: src/views/Swap/useLockOptions.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+export const MIN_LOCK_DURATION = "24";
+
+function useLockOptions() {
+  const [lockOptions, setLockOptions] = React.useState<Array<{ name: string; value: number }>>([]);
+  const { provider, chainID } = useWeb3Context();
+
+  React.useEffect(() => {
+    if (chainID && provider) {
+      (async function () {
```
Comment (jiggyBu) [inline src/views/Swap/useLockOptions.ts:18]: try/catch 😛

**Review by jiggyBu — APPROVED** [review]
Left few, rest looks good 👍

File: src/views/Swap/useLockOptions.ts (hunk 4 above)
Comment (bbernaz) [reply src/views/Swap/useLockOptions.ts:18]: will never happen again 😆

**Pull request merged**